pkgs/applications/networking/browsers/chromium/common.nix

```diff
@@ -287,25 +287,9 @@ let
     } // optionalAttrs pulseSupport {
       use_pulseaudio = true;
       link_pulseaudio = true;
-    } // optionalAttrs ungoogled {
-      chrome_pgo_phase = 0;
-      enable_hangout_services_extension = false;
-      enable_js_type_check = false;
-      enable_mdns = false;
-      enable_nacl_nonsfi = false;
-      enable_one_click_signin = false;
-      enable_reading_list = false;
-      enable_remoting = false;
-      enable_reporting = false;
-      enable_service_discovery = false;
-      exclude_unwind_tables = true;
-      google_api_key = "";
-      google_default_client_id = "";
-      google_default_client_secret = "";
-      safe_browsing_mode = 0;
-      use_official_google_api_keys = false;
-      use_unofficial_version_number = false;
-    } // (extraAttrs.gnFlags or {}));
+    })
+    + " " + optionalString ungoogled (readFile "${ungoogler}/flags.gn")
+    + " " + mkGnFlags (extraAttrs.gnFlags or {});
```

Comment:
Just a quick heads-up: Using strings looks like a bad idea. This was an attribute set and (if possible) it should remain one (e.g., important for overriding).

Reply:
I see, in that case I think we can just use importTOML to get our attributes, since the syntax is mostly (completely?) the same. From my quick testing it can correctly read both strings and lists from GN files:

```nix
nix-repl> :p fromTOML ''                
            a = [ "foo" ]   
            b = "bar"
          ''
{ a = [ "foo" ]; b = "bar"; }
```

#
It can read `flags.gn` correctly:
<details><summary>output of `lib.importTOML &ltpath to ungoogled-chromium&gt;/flags.gn`</summary>
<p>

```nix
{ build_with_tflite_lib = false; chrome_pgo_phase = 0; clang_use_chrome_plugins = false; disable_fieldtrial_testing_config = true; enable_hangout_services_extension = false; enable_js_type_check = false; enable_mdns = false; enable_mse_mpeg2ts_stream_parser = true; enable_nacl = false; enable_one_click_signin = false; enable_reading_list = false; enable_remoting = false; enable_reporting = false; enable_service_discovery = false; enable_widevine = true; exclude_unwind_tables = true; google_api_key = ""; google_default_client_id = ""; google_default_client_secret = ""; safe_browsing_mode = 0; treat_warnings_as_errors = false; use_official_google_api_keys = false; use_unofficial_version_number = false; }
```

</p>
</details>

Reply:
Yes, `importTOML` would be nice :)
Another thing that I've missed: We (AFAIK) cannot use `readFile "${ungoogler}/flags.gn"` (technically we can via IFD but it is basically forbidden since it will require building `ungoogler` for the evaluation of `ungoogled-chromium` - and it should be disabled on Hydra).

But we can still use that approach by simply putting `flags.gn` inside Nixpkgs (maybe `pkgs/applications/networking/browsers/chromium/ungoogled-flags.nix`?).
It'd be great if you could update `pkgs/applications/networking/browsers/chromium/update.py` accordingly (but I could also help you with that).

Reply:
Done!
It's been a long time since I've last written any Python, so I'd appreciate any feedback on the changes to `update.py`.